src/bsp/src/posix/src/BSP.cpp

```diff
@@ -1,6 +1,64 @@
-#include "bsp/BSP.h"
+#include "BSP.h"
+#include "ros/ros.h"
+#include <FreeRTOS.h>
+#include <FreeRTOSConfig.h>
+#include <bsp/BSPFactory.h>
+#include <hive_mind/ExampleMessage.h>
+#include <sstream>
+#include <task.h>
+#include <timers.h>
 
 BSP::BSP() = default;
-BSP::~BSP() = default;
+BSP::~BSP() {
+    if (m_rosNodeHandle != NULL) {
+        delete m_rosNodeHandle;
+    }
+}
 
-void BSP::initChip() {}
+/**
+ * @brief Task that kills FreeRTOS when ROS node is stopped
+ */
+void rosWatcher(void* param) {
+    (void)param;
+    const int loopRate = 100;
+
+    while (ros::ok()) {
+        vTaskDelay(loopRate);
+    }
+
+    vTaskEndScheduler();
+}
+
+void exampleTopicPublish(void* param) {
+    (void)param;
+    const int loopRate = 4000;
+    hive_mind::ExampleMessage msg = hive_mind::ExampleMessage();
+    msg.number = 0;
+    msg.text = "Hello World";
+
+    BSP* bsp = (BSP*)(BSPFactory::getBSP());
+    ros::Publisher publisher =
+        bsp->getRosNodeHandle()->advertise<hive_mind::ExampleMessage>("exampleTopic", 1000);
+
+    while (true) {
+        publisher.publish(msg);
+        msg.number++;
+
+        vTaskDelay(loopRate);
```

Comment:
I see here that you don't use `ros::spin()`. I guess this has to do with the fact that we use FreeRTOS in the HiveMind to orchestrate the tasks. Could this be an issue when we integrate this with the other ROS nodes?

Reply:
My bad, it got removed at some point during my tests. I've added a call to `ros::spinOnce()` in the `ros_watcher` task, so it is called periodically. It doesn't seem to be causing any issues so far.

The only thing I found that doesn't work is using a sleep directly from ros as in the tutorial, but we can use a `vTaskDelay` without any problems
```
ros::Rate r(10); // 10 hz
while (ros::ok())
{
  ... do some work ...
  r.sleep();
}
```